CHANGELOG.md

```diff
@@ -73,6 +73,9 @@ Adding a new version? You'll need three changes:
 
 - The `CombinedServices` feature gate is now enabled by default.
   [#4138](https://github.com/Kong/kubernetes-ingress-controller/pull/4138)
+- Gateway resources no longer use the _Ready_ condition following changes to
+  the upstream Gateway API specification in version 0.7.
+  [#4142](https://github.com/Kong/kubernetes-ingress-controller/pull/4142)
```

Comment:
I really hate to be nitpicky but to me as a user, it would be best if I got the bump information verbatim. Ideally mentioning some noteworthy highlights of that release.

PTAL at the 0.5.1 and 0.6.1 examples that we already have in the changelog:

- https://github.com/Kong/kubernetes-ingress-controller/blob/71a1e09ba2de7027f987ed149db0dc0cbb2fdcda/CHANGELOG.md?plain=1#L295-L299
- https://github.com/Kong/kubernetes-ingress-controller/blob/71a1e09ba2de7027f987ed149db0dc0cbb2fdcda/CHANGELOG.md?plain=1#L548-L555

How about we try to use the same format and use something like:

```
- Gateway API has been bumped to 0.7.1.
  The Gateway `Ready` condition has been changed to `Programmed`.
```

and perhaps add some other relevant changes that I forgot to mention. WDYT?

Reply:
IMO the change users are going to care about is what they actually see when working with resources, which is that Ready is no longer in use, with most of the finer details in the linked PR.

We generally _don't_ list library bumps themselves. The "because upstream" note here is more because we want some explanation of why we did it. Leaving as is.

Reply:
Let's leave it as is then but I'm still hesitant that the format we've already used would be clearer for the reader (especially since we're bumping to 0.7.1 not 0.7).